src/nvim/macros.h

```diff
@@ -203,16 +205,31 @@
 # define PRAGMA_DIAG_PUSH_IGNORE_MISSING_PROTOTYPES \
   _Pragma("clang diagnostic push") \
   _Pragma("clang diagnostic ignored \"-Wmissing-prototypes\"")
+# ifdef HAVE_WIMPLICIT_FALLTHROUGH_FLAG
+#  define PRAGMA_DIAG_PUSH_IGNORE_IMPLICIT_FALLTHROUGH \
+    _Pragma("clang diagnostic push") \
+    _Pragma("clang diagnostic ignored \"-Wimplicit-fallthrough\"")
+# else
```

Comment:
any reason why we dont define PRAGMA_DIAG_PUSH_IGNORE_IMPLICIT_FALLTHROUGH via a `#cmakedefine` instead of preproc ? this could get rid of HAVE_WIMPLICIT_FALLTHROUGH_FLAG

Reply:
> any reason why we dont define PRAGMA_DIAG_PUSH_IGNORE_IMPLICIT_FALLTHROUGH via a `#cmakedefine` instead of preproc ?

I tried that and it's more complicated than it's worth.  `HAVE_WIMPLICIT_FALLTHROUGH_FLAG` does the job.